week2/touching_properties.ipynb

```diff
@@ -0,0 +1,216 @@
+    "    m.fs.unit.properties[0].conc_mass_phase_comp[\"Liq\", \"TDS\"]\n",
+    "```\n",
+    "\n",
+    "Importantly, touching property variables is not necessary for unit model state blocks that use the property. For example, the RO model requires the use of `pressure_osm_phase` on the *inlet* stream so you would not need to touch that property for that state block. *But*, if you were interested in knowing the osmotic pressure of the permeate stream, you would need to touch the property on the permeate state block prior to solving."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9bc2128d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pyomo.environ import ConcreteModel, value\n",
+    "\n",
+    "from idaes.core import FlowsheetBlock\n",
+    "from idaes.models.unit_models import Feed\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "\n",
+    "from watertap.property_models.seawater_prop_pack import SeawaterParameterBlock\n",
+    "\n",
+    "from watertap.core.solvers import get_solver\n",
+    "\n",
+    "solver = get_solver()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "065a6c1f",
+   "metadata": {},
+   "source": [
+    "## Two Build Functions\n",
+    "\n",
+    "Below are two build functions. One touches both `pressure_osm_phase` and `conc_mass_phase_comp` prior to solving. The other only touches `conc_mass_phase_comp`. \n",
+    "\n",
+    "Running this cell will create, initialize, and solve both models, and then print out the concentration and osmotic pressure for both."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c07cf9dc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def run_build_with_touch():\n",
+    "    \"\"\"\n",
+    "    Create feed model using seawater property package\n",
+    "    Touch properties prior to solving\n",
+    "    \"\"\"\n",
+    "    m = ConcreteModel()\n",
+    "    m.fs = FlowsheetBlock(dynamic=False)\n",
+    "    m.fs.properties = SeawaterParameterBlock()\n",
+    "    m.fs.feed = Feed(property_package=m.fs.properties)\n",
+    "    m.fs.feed.properties[0].pressure_osm_phase\n",
+    "    m.fs.feed.properties[0].conc_mass_phase_comp\n",
+    "    m.fs.feed.properties[0].flow_mass_phase_comp[\"Liq\", \"H2O\"].fix(0.965)\n",
+    "    m.fs.feed.properties[0].flow_mass_phase_comp[\"Liq\", \"TDS\"].fix(0.035)\n",
+    "    m.fs.feed.properties[0].temperature.fix(273 + 25)\n",
+    "    m.fs.feed.properties[0].pressure.fix(101325)\n",
+    "\n",
+    "    print(f\"Model 1 degrees of freedom: {degrees_of_freedom(m)}\\n\")\n",
+    "\n",
+    "    # Initialize and solve for the initial conditions\n",
+    "    m.fs.feed.initialize()\n",
+    "    results = solver.solve(m)\n",
+    "    print(f\"Model 1 termination {results.solver.termination_condition}\")\n",
+    "\n",
+    "    return m\n",
+    "\n",
+    "\n",
+    "def run_build_without_touch():\n",
```

Comment:
This name is no longer accurate - consider changing this and the name for the function above